### PR TITLE
fix(stackdriver): map grpc status codes into http domain for metrics

### PR DESCRIPTION
**Pull request by douglas-reid**
Adds logic for stackdriver metrics reporting to handle gRPC status codes by translating to HTTP response codes. This should allow monitoring of gRPC services to work seamlessly in HTTP-focused dashboards and alerting.

### CODE REVIEW

File: extensions/stackdriver/metric/record.cc [hunk 1]
```diff
@@ -169,15 +169,63 @@ TagKeyValueList getInboundTagMap(
   return inboundMap;
 }
 
+// See:
+// https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto
+uint32_t httpCodeFromGrpc(uint32_t grpc_status) {
+  switch (grpc_status) {
+    case 0:
```
Comment (gargnupur) [inline extensions/stackdriver/metric/record.cc:176]: Can we add the status code names here?

Comment (douglas-reid) [reply extensions/stackdriver/metric/record.cc:176]: do you mean in the code or in additional comments?

all we have is the uint32_t (in RequestInfo) unfortunately. we could convert and then convert again, but that seems a bit wasteful.

Comment (gargnupur) [reply extensions/stackdriver/metric/record.cc:176]: sorry meant just in comments....

Comment (douglas-reid) [reply extensions/stackdriver/metric/record.cc]: Done.

**Comment by douglas-reid** [comment]
@gargnupur PTAL. Updated to add comments as you suggested.

**Comment by douglas-reid** [comment]
@mandarjog do we need to cherry-pick this back to 1.6 ?

**Comment by mandarjog** [comment]
It is a bug fix of sorts, we should put it back in 1.6.

**Pull request merged**